### PR TITLE
Skip iconv for problematic ISO-2022-JP strings 

### PR DESCRIPTION
**Pull request by ernix**
ref: https://github.com/roundcube/roundcubemail/pull/5665

### CODE REVIEW

File: tests/Framework/Charset.php [hunk 1]
```diff
@@ -64,13 +64,27 @@ function test_parse_charset($input, $output)
...
-        return array(
+        $tests = array(
             array('ö', 'ö', 'UTF-8', 'UTF-8'),
             array('ö', '', 'UTF-8', 'US-ASCII'),
             array('aż', 'a', 'UTF-8', 'US-ASCII'),
             array('&BCAEMARBBEEESwQ7BDoEOA-', 'Рассылки', 'UTF7-IMAP', 'UTF-8'),
             array('Рассылки', '&BCAEMARBBEEESwQ7BDoEOA-', 'UTF-8', 'UTF7-IMAP'),
         );
+
+        if (extension_loaded('mbstring')
```
Comment (alecpl) [inline tests/Framework/Charset.php]: As I said in another place, this is required extension. No need to check it here.

File: program/lib/Roundcube/rcube_charset.php [hunk 2]
```diff
@@ -200,8 +200,30 @@ public static function convert($str, $from, $to = null)
...
         }
 
+        $aliases = array(
+            'WINDOWS-1257' => 'ISO-8859-13',
+            'US-ASCII'     => 'ASCII',
+        );
+
         // convert charset using iconv module
-        if ($iconv_options !== false && $from != 'UTF7-IMAP' && $to != 'UTF7-IMAP') {
+        do {
```
Comment (alecpl) [inline program/lib/Roundcube/rcube_charset.php]: I think you made this code overcomplicated. I would change only the if statement. I don't like do/while and mb_check_encoding().

**Comment by ernix** [comment]
> I don't like do/while and mb_check_encoding().

OK, I removed do/while and mb_check_encoding.

-- 
Shin Kojima

**Comment by ernix** [comment]
> No need to check it here.

OK, I kept mb_substitute_character as current code from `rcube_charset::convert`

    223         if ($mbstring_sc === null) {
    224             $mbstring_sc = extension_loaded('mbstring') ? mb_substitute_character() : false;
    225         }

-- 
Shin Kojima

**Comment by ernix** [comment]
@alecpl Should I remove `mb_substitute_character` check in the test too?

**Comment by alecpl** [comment]
Yes, it does not make sense.

**Comment by ernix** [comment]
@alecpl done.  Is it acceptable for you?

**Pull request merged**